src/randoop/util/ListOfLists.java

```diff
@@ -95,4 +98,44 @@ public T get(int index) {
   public String toString() {
     return toJDKList().toString();
   }
+
+  @Override
+  public Iterator<T> iterator() {
+    return new LOLIterator(lists.iterator());
+  }
+  
+  private class LOLIterator implements Iterator<T> {
+
+    private Iterator<SimpleList<T>> listIterator;
+    private Iterator<T> elemIterator;
+
+    public LOLIterator(Iterator<SimpleList<T>> listIterator) {
+      this.listIterator = listIterator;
+      if (this.listIterator.hasNext()) {
+        this.elemIterator = (this.listIterator).next().iterator();
+      }
+    }
+
+    @Override
+    public boolean hasNext() {
+      return (listIterator.hasNext() || (elemIterator != null && elemIterator.hasNext()));
+    }
+
+    @Override
+    public T next() {
+      if (elemIterator != null) {
+        if (!elemIterator.hasNext()) {
+          elemIterator = listIterator.next().iterator();
+        }
+        return elemIterator.next();
+      }
+      throw new NoSuchElementException("end of ListOfLists reached");
+    }
+    
+    @Override
+    public void remove() {
+      //does nothing
```

Comment:
If this has no effect, its body should be `throw new UnsupportedOperationException();`.